api_app/playbooks_manager/controller.py

```diff
@@ -0,0 +1,150 @@
+# This file is a part of IntelOwl https://github.com/intelowlproject/IntelOwl
+# See the file 'LICENSE' for copying permission.
+
+import logging
+from typing import Dict, List, Tuple
+
+from celery import chord
+from django.conf import settings
+
+from api_app.exceptions import (
+    NotRunnablePlaybook,
+)
+from ..utility import job_cleanup, runnable_connectors, stack_analyzers, stack_connectors, runnable_analyzers
+
+from ..models import Job
+from .dataclasses import PlaybookConfig
+
+logger = logging.getLogger(__name__)
+
+def filter_playbooks(serialized_data: Dict) -> Tuple[List[str]]:
+    # init empty list
+    valid_playbook_list = []
+    selected_playbooks = []
+    analyzers_to_be_run = []
+    connectors_to_be_run = []
+    warnings = []
+
+    # get values from serializer
+    playbooks_requested = serialized_data[0].get("playbooks_requested", [])
+
+    # read config
+    playbook_dataclasses = PlaybookConfig.all()
+    all_playbook_names = list(playbook_dataclasses.keys())
+
+    # run all playbooks; Might just remove this for now.
+    run_all = len(playbooks_requested) == 0
+    if run_all:
+        # select all
+        selected_playbooks.extend(all_playbook_names)
+    else:
+        # select the ones requested
+        selected_playbooks.extend(playbooks_requested)
+
+    for p_name in selected_playbooks:
+        try:
+            pp = playbook_dataclasses.get(p_name, None)
+            if not pp:
+                if not run_all:
+                    raise NotRunnablePlaybook(
+                        f"{p_name} won't run: not present in configuration"
+                    )
+                continue
+            if not pp.is_ready_to_use:
+                raise NotRunnablePlaybook(
+                    f"{p_name} won't run: not configured"
+                )
+
+            playbook_analyzers = runnable_analyzers(pp.analyzers)
+            playbook_connectors = runnable_connectors(pp.connectors)
```

Comment:
Not necessary assignment

Reply:
I did it especially thinking that it would make it more readable. But if you suggest so, I would get to remove it.

Reply:
I get your point, and this is trivial and just my own preference. If you think that is more readable as you written, i'm 100% ok with keeping that way